src/main/kotlin/io.javalin/embeddedserver/jetty/EmbeddedJettyServer.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin.embeddedserver.jetty
+
+import io.javalin.Javalin
+import io.javalin.embeddedserver.EmbeddedServer
+import org.eclipse.jetty.server.Connector
+import org.eclipse.jetty.server.Server
+import org.eclipse.jetty.server.ServerConnector
+import org.eclipse.jetty.server.session.SessionHandler
+import org.slf4j.LoggerFactory
+import java.io.IOException
+import java.net.ServerSocket
+
+class EmbeddedJettyServer(private var server: Server, private val javalinHandler: SessionHandler) : EmbeddedServer {
+
+    private val log = LoggerFactory.getLogger(EmbeddedServer::class.java)
+
+    @Throws(Exception::class)
+    override fun start(host: String, port: Int): Int {
+        var portVar = port
+
+        if (portVar == 0) {
+            try {
+                ServerSocket(0).use { serverSocket -> portVar = serverSocket.localPort }
+            } catch (e: IOException) {
+                log.error("Failed to get first available port, using default port instead: " + Javalin.DEFAULT_PORT)
+                portVar = Javalin.DEFAULT_PORT
+            }
+
+        }
```

Comment:
this still bugs me a bit - i think you could clean it up some with expression returns like so:

```kotlin
val portVar = if (port == 0) {
             try {
                 ServerSocket(0).use { it.localPort }
             } catch (e: IOException) {
                 log.error("Failed to get first available port, using default port instead: " + Javalin.DEFAULT_PORT)
                Javalin.DEFAULT_PORT
             }
          } else {
                port
          }
```